ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -321,6 +332,27 @@ func createCluster(context *cli.Context, awsClients *AWSClients, commandConfig *
 	return cfnClient.WaitUntilCreateComplete(stackName)
 }
 
+func determineArchitecture(cfnParams *cloudformation.CfnStackParams) (string, error) {
+	architecture := amimetadata.ArchitectureTypeX86
+
+	// a1 instances get the Arm based ECS AMI
+	instanceTypeParam, err := cfnParams.GetParameter(ParameterKeyInstanceType)
+	if err == cloudformation.ParameterNotFoundError {
+		logrus.Infof("Defaulting instance type to t2.micro")
+	} else if err != nil {
+		return "", err
+	} else {
+		instanceType := aws.StringValue(instanceTypeParam.ParameterValue)
+		r := regexp.MustCompile("a1\\.(medium|\\d*x?large)")
```

Comment:
Maybe this is too pedestrian, but I don't like regex if it can be avoided:

    strings.HasPrefix(instanceType, "a1")

Reply:
Yeah, I started with this, but @SoManyHs felt it was a tad too permissive.
#codestyleisasocialconstruct